src/core/mdn-annotation.js

```diff
@@ -0,0 +1,170 @@
+  // Find the furthest ancestor that is a direct child of <body>
+  while (targetAncestor.parentNode.tagName !== "BODY") {
+    targetAncestor = targetAncestor.parentNode;
+  }
+  const targetSibling = targetAncestor.previousElementSibling;
+  if (targetSibling && targetSibling.classList.contains("mdn")) {
+    // If the target ancestor already has a mdnBox inserted, we just use it
+    return targetSibling;
+  }
+  const mdnBox = hyperHTML`<aside class="mdn before">
+    <input type="button" onclick="toggleStatus(this)" value="⋰" >
+  </aside>`;
+  document.body.insertBefore(mdnBox, targetAncestor);
+  return mdnBox;
+}
+
+function attachMDNDetail(container, mdnSpec) {
+  const { slug, summary } = mdnSpec;
+  container.innerHTML += `<b>MDN </b>`;
+  const mdnSubPath = slug.slice(slug.indexOf("/") + 1);
+  const mdnDetail = document.createElement("details");
+  mdnDetail.innerHTML += `
+    <summary>
+      <a title="${summary}" href="${MDN_URL_BASE}${slug}">${mdnSubPath}</a>
+    </summary>
+  `;
+  attachMDNBrowserSupport(mdnDetail, mdnSpec);
+  container.appendChild(mdnDetail);
+}
+
+function attachMDNBrowserSupport(container, mdnSpec) {
+  if (!mdnSpec.support) {
+    container.innerHTML += `<p class="nosupportdata">No support data.</p>`;
+    return;
+  }
+  const supportTable = hyperHTML`<p class="mdnsupport"></p>`;
+  buildBrowserSupportTable(supportTable, mdnSpec.support);
+  container.appendChild(supportTable);
+}
+
+function buildBrowserSupportTable(supportTable, support) {
+  function addMDNBrowserRow(browserId, yesNoUnknown, version) {
+    const displayStatus = yesNoUnknown === "Unknown" ? "?" : yesNoUnknown;
+    const classList = `${browserId} ${yesNoUnknown.toLowerCase()}`;
+    const browserRow = hyperHTML`
+      <span class="${classList}">
+        <span>${MDN_BROWSERS[browserId]}</span>
+        <span>${version ? version : displayStatus}</span>
+      </span>`;
+    supportTable.appendChild(browserRow);
+  }
+
+  function processBrowserData(browserId, versionData) {
+    if (versionData.version_removed) {
+      addMDNBrowserRow(browserId, "No", "");
+      return;
+    }
+    const versionAdded = versionData.version_added;
+    if (!versionAdded) {
+      addMDNBrowserRow(browserId, "Unknown", "");
+      return;
+    }
+    if (typeof versionAdded === "boolean") {
+      addMDNBrowserRow(browserId, versionAdded ? "Yes" : "No", "");
+    } else {
+      addMDNBrowserRow(browserId, "Yes", `${versionAdded}+`);
+    }
+  }
+
+  Object.keys(MDN_BROWSERS).forEach(browserId => {
+    if (!support[browserId]) {
+      addMDNBrowserRow(browserId, "Unknown", "");
+    } else {
+      if (Array.isArray(support[browserId])) {
+        support[browserId].forEach(b => {
+          processBrowserData(browserId, b);
+        });
+      } else {
+        processBrowserData(browserId, support[browserId]);
+      }
+    }
+  });
+}
+
+export async function run(conf) {
+  const { shortName, mdnAnnotation } = conf;
+  if (!shortName) {
```

Comment:
So, we should only add the mdn box if `mdnAnnotation` is set...
 
```suggestion
  if (!shortName || !mdnAnnotation) {
```

Also, we probably just want to call it `mdn` as "Annotation" is hard to spell.

Reply:
That's a good idea. I find it hard to spell too...